nntrainer/models/neuralnet.cpp

```diff
@@ -301,6 +301,18 @@ sharedConstTensors NeuralNetwork::forwarding(sharedConstTensors input,
   auto &last_layer =
     model_graph.getSortedLayerNode(model_graph.getSorted().size() - 1).layer;
 
+  for (auto &layer_node : model_graph.getSorted()) {
+    auto l = layer_node.layer;
+    if (l->getType() == "centroid_nn") {
+      if (!label.empty()) {
+        throw std::invalid_argument(
+          "centroid nn forwarding requires label feeded");
+      }
+
+      l->net_hidden[0]->getGradientRef() = *label[0].get();
```

Comment:
Also, there must a check that knn is the last layer (and now loss layer is not the only layer which can be last). Please add that check as well.

Reply:
Currently loss layer is added, this should be fixed with mechanism always adding loss layer. Instead added comment